pkg/reconcilers/marin3r/envoyconfigrevision/cache.go

```diff
@@ -138,19 +138,19 @@ func (r *CacheReconciler) GenerateSnapshot(req types.NamespacedName, resources *
 			Namespace: secret.Ref.Namespace,
 		}
 		if err := r.client.Get(r.ctx, key, s); err != nil {
-			return nil, err
+			return nil, fmt.Errorf("%s", err.Error())
```

Comment:
This is actually what does the trick and solves the problem.  I am reusing the k8s error type in this controller so it happens that generic errors like NotFount on get operations are also of the same type, which I didn't realise back then when I went down that path. If you check the main controller code [here](https://github.com/3scale/marin3r/blob/33f9ae0f036fd5518ebf59c7ed2d577537c6911b/controllers/marin3r/envoyconfigrevision_controller.go#L116-L125) you will see that depending on the error type returned, the revision is tainted or not, which was not working because generic k8s API error are also of the same type.
To avoid that I am forcing it to return a generic error so the type switch works as expected. This is a hacky way to fix it, and the better solution would be to create our own error type, but for now it solves the problem, though not in a very elegant way.

Reply:
👌 🧙 ✨